src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -90,10 +98,18 @@
    * @return completable future with {@link CompositePurchaseOrder} object with populated uuid on success or an exception if processing fails
    */
   public CompletableFuture<CompositePurchaseOrder> createPurchaseOrder(CompositePurchaseOrder compPO) {
-    return setPoNumberIfMissing(compPO)
-      .thenCompose(v -> poNumberHelper.checkPONumberUnique(compPO.getPoNumber()))
-      .thenCompose(v -> createPOandPOLines(compPO))
-      .thenApply(this::populateOrderSummary);
+    return protectionHelper.isOperationRestricted(compPO.getAcqUnitIds())
+      .thenCompose(isProtected -> {
+        if(isProtected) {
+          throw new HttpException(HttpStatus.HTTP_FORBIDDEN.toInt(), USER_HAS_NOT_PERMISSIONS);
```

Comment:
The result of the `isOperationRestricted` method is just used to throw in an exception in case the operation is restricted. It may be better to throw an exception inside the method when the operation is restricted otherwise just to return `CompleatableFuture<Void>`